plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetUnmanagedInstancesCommandWrapper.java

```diff
@@ -203,14 +197,17 @@ private List<UnmanagedInstanceTO.Disk> getUnmanagedInstanceDisks(List<LibvirtVMD
                 disk.setDatastoreHost(sourceHostPath.first());
                 disk.setDatastorePath(sourceHostPath.second());
             } else {
-                disk.setDatastorePath(diskDef.getSourcePath());
+                int pathEnd = diskDef.getSourcePath().lastIndexOf("/");
+                if (pathEnd >= 0) {
+                    disk.setDatastorePath(diskDef.getSourcePath().substring(0, pathEnd));
+                }
                 disk.setDatastoreHost(diskDef.getSourceHost());
             }
 
             disk.setDatastoreType(diskDef.getDiskType().toString());
             disk.setDatastorePort(diskDef.getSourceHostPort());
-            disk.setImagePath(imagePath);
-            disk.setDatastoreName(imagePath.substring(imagePath.lastIndexOf("/")));
+            disk.setImagePath(diskDef.getSourcePath());
+            disk.setDatastoreName(disk.getDatastorePath());
```

Comment:
@slavkap I tested your PR on advanced zone on NFS storage and was able to import and list correctly (no regressions). However I found some differences:

In case of NFS the datastore name is set incorrectly as the disk's relative path on the pool (previously was the same as the datastore path):
````
(localcloud) 🐱 > list unmanagedinstances clusterid=e4ac8607-6839-42ef-abb9-fa66d3512b3a
{
  "count": 1,
  "unmanagedinstance": [
    {
      "clusterid": "e4ac8607-6839-42ef-abb9-fa66d3512b3a",
      "clustername": "p1-c2",
      "cpucorepersocket": -1,
      "cpunumber": 1,
      "cpuspeed": 397,
      "disk": [
        {
          "capacity": 8589934592,
          "controller": "virtio",
          "datastorehost": "10.0.32.4",
          "datastorename": "/ea4b2296-d349-4968-ab72-c8eb523b556e",
          "datastorepath": "/acs/primary/ref-trl-6074-k-M7-nicolas-vazquez/ref-trl-6074-k-M7-nicolas-vazquez-kvm-pri1",
          "datastoretype": "file",
          "id": "0",
          "imagepath": "/mnt/97e4c9ed-e3bc-3e26-b103-7967fc9feae1/ea4b2296-d349-4968-ab72-c8eb523b556e",
          "label": "vda",
          "position": 0
        }
      ],
````

From the VM XML domain:
````
    <disk type='file' device='disk'>
      <driver name='qemu' type='qcow2' cache='none'/>
      <source file='/mnt/97e4c9ed-e3bc-3e26-b103-7967fc9feae1/ea4b2296-d349-4968-ab72-c8eb523b556e' index='2'/>
````

Reply:
Sorry, confused the jars while testing. Its working properly after the fix